src/helpers.ts

```diff
@@ -0,0 +1,28 @@
+import { log, BigInt, Bytes } from "@graphprotocol/graph-ts";
+import { StoreCache } from "./store-cache";
+import { TUPLE_OFFSET_HEX } from './constants';
+
+export function parseCalldata(calldata: Bytes): Bytes {
+  // Remove function signature
+  let dataWithoutSignature = calldata.toHexString().slice(10);
+  let result = TUPLE_OFFSET_HEX + dataWithoutSignature;
```

Comment:
The tuple offset fix is not very intuitive so I would add either a link to the article explaining it (https://medium.com/@r2d2_68242/indexing-transaction-input-data-in-a-subgraph-6ff5c55abf20) or an explanation as comments.